api/provider/aws/load_balancer_delete.go

```diff
@@ -44,6 +46,51 @@ func (l *LoadBalancerProvider) Delete(loadBalancerID string) error {
 		}
 	}
 
+	// Confirm Delete
+	consistencyCheckFN := func() error {
+		// Check IAM Role
+		getRoleInput := &iam.GetRoleInput{}
+		getRoleInput.SetRoleName(roleName)
+
+		if _, err := l.AWS.IAM.GetRole(getRoleInput); err != nil {
+			if err, ok := err.(awserr.Error); ok && err.Code() == "NoSuchEntity" {
+				return nil
+			}
+
+			return err
+		}
+
+		// Check Role Policy
+		getRolePolicyInput := &iam.GetRolePolicyInput{}
+		getRolePolicyInput.SetRoleName(roleName)
+		getRolePolicyInput.SetPolicyName(policyName)
+
+		if _, err := l.AWS.IAM.GetRolePolicy(getRolePolicyInput); err != nil {
+			if err, ok := err.(awserr.Error); ok && err.Code() == "NoSuchEntity" {
+				return nil
+			}
+
+			return err
+		}
+
+		// Check Security Group
+		_, err := readSG(l.AWS.EC2, securityGroupName)
+		if err != nil && !strings.Contains(err.Error(), "does not exist") {
+			return err
+		}
+
+		_, err = describeLoadBalancer(l.AWS.ELB, fqLoadBalancerID)
+		if serverError, ok := err.(*errors.ServerError); ok && serverError.Code == "LoadBalancerDoesNotExist" {
+			return nil
+		}
+
+		return err
+	}
+
+	if err := retry(config.DefaultRetryTimeout, config.DefaultRetryTick, consistencyCheckFN); err != nil {
```

Comment:
how do we know when to retry and when there is a fatal error?

Reply:
What determines a fatal error? Should it be with 1 or more errors?

Reply:
fatal probably isn't the best word: how do we determine when to retry and when to not retry in the face of errors?